sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/rule/EncryptRule.java

```diff
@@ -132,7 +132,18 @@ public String getLogicColumnOfCipher(final String logicTable, final String ciphe
      * @return plain column
      */
     public Optional<String> findPlainColumn(final String logicTable, final String logicColumn) {
-        return tables.containsKey(logicTable) ? tables.get(logicTable).findPlainColumn(logicColumn) : Optional.<String>absent();
+        return tables.containsKey(logicTable) ? tables.get(logicTable).findPlainColumn(getOriginColumnName(logicTable, logicColumn)) : Optional.<String>absent();
+    }
+
+    private String getOriginColumnName(final String logicTable, final String logicColumn) {
+        String result = logicColumn;
+        for (String each : tables.get(logicTable).getLogicColumns()) {
+            if (logicColumn.equals(each.toLowerCase())) {
+                result = each;
+                break;
+            }
+        }
+        return result;
```

Comment:
What is situation if `logicColumn` cannot find? Maybe throw exception?

Reply:
If the logicColum cannot be found, just return the original parameter as logic parameter, it work.

Reply:
Hi, thank you for reply quickly.
I know return  original parameter will not break the code. I just want ti know, what is situation if logicColumn cannot find? Is it impossible? If yes, shall we use `absent` or throw exception to make the code more clear?

Reply:
Of course it is possible. If the logicColumn can not be found as well as the encryptor, it is normal which means we do not need to encrypt the column. See the source code methods by this step:
1.1.1  org.apache.shardingsphere.sql.rewriter.encrypt.parameter.impl.EncryptInsertValueParameterRewriter#rewrite
1.1.2
org.apache.shardingsphere.sql.rewriter.encrypt.parameter.impl.EncryptAssignmentParameterRewriter#rewrite
1.2
org.apache.shardingsphere.core.rule.EncryptRule#findShardingEncryptor(String,String)

Reply:
So, how about return `Optional`? If logicColumn cannot find, just return `Optional.absent()`.
And do not need call `findPlainColumn()` if absent returned.

Reply:
It make sense. Let me fix it.